plugin/core/configurations.py

```diff
@@ -112,26 +73,29 @@ def is_supported(self, view: Any) -> bool:
         return any(self.scope_configs(view))
 
     def scope_configs(self, view: Any, point: Optional[int] = None) -> Iterator[ClientConfig]:
-        return get_scope_client_configs(view, self.all, point)
+        scope = view2scope(view, point)
+        for config in self.all:
+            if config.supports(scope):
+                yield config
 
     def syntax_configs(self, view: Any, include_disabled: bool = False) -> List[ClientConfig]:
-        syntax = view.settings().get("syntax")
-        return list(filter(lambda c: config_supports_syntax(c, syntax) and (c.enabled or include_disabled), self.all))
+        scope = view2scope(view)
+        return list(filter(lambda c: c.supports(scope) and (c.enabled or include_disabled), self.all))
 
-    def syntax_supported(self, view: ViewLike) -> bool:
-        syntax = view.settings().get("syntax")
-        for found in filter(lambda c: config_supports_syntax(c, syntax) and c.enabled, self.all):
+    def syntax_supported(self, view: sublime.View) -> bool:
+        scope = view2scope(view)
+        for found in filter(lambda c: c.supports(scope) and c.enabled, self.all):
             return True
         return False
 
-    def syntax_config_languages(self, view: ViewLike) -> Dict[str, LanguageConfig]:
-        syntax = view.settings().get("syntax")
+    def syntax_config_languages(self, view: sublime.View) -> Dict[str, LanguageConfig]:
+        scope = view2scope(view)
         config_languages = {}
         for config in self.all:
             if config.enabled:
-                language = syntax_language(config, syntax)
-                if language:
-                    config_languages[config.name] = language
+                for language in config.languages:
+                    if language.score(scope) > 0:
```

Comment:
Can we use `match` here? The difference would be that this can return True also for `1` but I don't think that case is possible because we can not have empty selector due to this logic:
```
self.selector = selector if selector else "source.{}".format(self.id)
```

(We could then even make `score` method private as it wouldn't be needed anymore, I think.)